bycycle/group/features.py

```diff
@@ -0,0 +1,228 @@
+"""Functions to compute features across 2 dimensional arrays of data."""
+
+from functools import partial
+from multiprocessing import Pool, cpu_count
+
+import numpy as np
+
+from bycycle.features import compute_features
+from bycycle.group.utils import progress_bar
+
+###################################################################################################
+###################################################################################################
+
+def compute_features_2d(sigs, fs, f_range, compute_features_kwargs=None,
+                        return_samples=True, n_jobs=-1, progress=None):
+    """Compute shape and burst features for a 2 dimensional array of signals.
+
+    Parameters
+    ----------
+    sigs : 2d array
+        Voltage time series, with shape [n_signals, n_points].
+    fs : float
+        Sampling rate, in Hz.
+    f_range : tuple of (float, float)
+        Frequency range for narrowband signal of interest, in Hz.
+    compute_features_kwargs : dict or list of dict
+        Keyword arguments used in :func:`~.compute_features`.
+    return_samples : bool, optional, default: True
+        Whether to return a dataframe of cyclepoint sample indices.
+    n_jobs : int, optional, default: -1
+        The number of jobs, one per cpu, to compute features in parallel.
+    progress: {None, 'tqdm', 'tqdm.notebook'}, optional, default: None
+        Specify whether to display a progress bar. Use 'tqdm' if installed.
+
+    Returns
+    -------
+    df_features : list of pandas.DataFrame
+        Dataframes containing shape and burst features for each cycle.
+        Each dataframe is computed using the :func:`~.compute_features` function.
+    df_samples : list of pandas.DataFrame, optional
+        Dataframes containing cyclepoints for each cycle.
+        Only returned if ``return_samples`` is True.
+        Each dataframe is computed using the :func:`~.compute_features` function.
+
+    Notes
+    -----
+
+    - The order of ``df_features`` and ``df_samples`` corresponds to the order of ``sigs``.
+    - If ``compute_features_kwargs`` is a dictionary, the same kwargs are applied applied across
+      the first axis of ``sigs``. Otherwise, a list of dictionaries equal in length to the
+      first axis of ``sigs`` is required to apply unique kwargs to each signal.
+    - ``return_samples`` is controlled from the kwargs passed in this function. If
+      ``return_samples`` is a key in ``compute_features_kwargs``, it's value will be ignored.
+
+    """
+
+    if isinstance(compute_features_kwargs, list) and len(compute_features_kwargs) != len(sigs):
+        raise ValueError("When compute_features_kwargs is a list, it's length must be equal to "
+                         "sigs. Use a dictionary when applying the same kwargs to each signal.")
+    elif compute_features_kwargs is None:
+        compute_features_kwargs = {}
+
+    # Remove return_samples from compute_features_kwargs
```

Comment:
Thanks for this! I totally agree. I went with this to keep line lengths shorter and to avoid aliases:
```
compute_features_kwargs = [compute_features_kwargs] if \
    isinstance(compute_features_kwargs, dict) else compute_features_kwargs
[kwargs.pop('return_samples', None) for kwargs in compute_features_kwargs]
```